cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/runtime/spi/provisioner/TetheringConf.java

```diff
@@ -0,0 +1,128 @@
+ * the License.
+ */
+
+package io.cdap.cdap.internal.tethering.runtime.spi.provisioner;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.PrivateModule;
+import com.google.inject.Scopes;
+import io.cdap.cdap.api.metrics.MetricsCollectionService;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.guice.ConfigModule;
+import io.cdap.cdap.data.runtime.StorageModule;
+import io.cdap.cdap.data.runtime.SystemDatasetRuntimeModule;
+import io.cdap.cdap.data.runtime.TransactionExecutorModule;
+import io.cdap.cdap.internal.tethering.NamespaceAllocation;
+import io.cdap.cdap.internal.tethering.PeerInfo;
+import io.cdap.cdap.internal.tethering.TetheringStatus;
+import io.cdap.cdap.internal.tethering.TetheringStore;
+import io.cdap.cdap.metrics.collect.LocalMetricsCollectionService;
+import io.cdap.cdap.spi.data.transaction.TransactionRunner;
+import org.apache.tephra.runtime.TransactionModules;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Configuration for the Tethering Provisioner.
+ */
+public class TetheringConf {
+  public static final String TETHERED_INSTANCE_PROPERTY = "tetheredInstanceName";
+  public static final String TETHERED_NAMESPACE_PROPERTY = "tetheredNamespace";
+
+  private final String tetheredInstanceName;
+  private final String tetheredNamespace;
+
+  private TetheringConf(String tetheredInstanceName, String tetheredNamespace) {
+    this.tetheredInstanceName = tetheredInstanceName;
+    this.tetheredNamespace = tetheredNamespace;
+  }
+
+  /**
+   * Create the conf from a property map while also performing validation that tethered connection exists.
+   */
+  public static TetheringConf fromProperties(Map<String, String> properties) {
+    String tetheredInstanceName = getString(properties, TETHERED_INSTANCE_PROPERTY);
+    String tetheredNamespace = getString(properties, TETHERED_NAMESPACE_PROPERTY);
+
+    Injector injector = createInjector();
+    TetheringStore tetheringStore = new TetheringStore(injector.getInstance(TransactionRunner.class));
+    checkTetheredConnection(tetheringStore, tetheredInstanceName, tetheredNamespace);
+
+    return create(tetheredInstanceName, tetheredNamespace);
+  }
+
+  /**
+   * Create the conf without performing validation.
+   */
+  @VisibleForTesting
+  public static TetheringConf create(String tetheredInstanceName, String tetheredNamespace) {
+    return new TetheringConf(tetheredInstanceName, tetheredNamespace);
+  }
+
+  private static String getString(Map<String, String> properties, String key) {
+    String val = properties.get(key);
+    if (val == null) {
+      throw new IllegalArgumentException(String.format("Invalid config. '%s' must be specified.", key));
+    }
+    return val;
+  }
+
+  @VisibleForTesting
+  static void checkTetheredConnection(TetheringStore store, String tetheredInstanceName, String tetheredNamespace) {
```

Comment:
why do we need this check?

Reply:
Removed the check since 1) existing provisioners do not do a similar type of validation and 2) the tethered connection state can change afterwards and the profile will no longer represent the accurate status